zerocopy-derive/tests/ui-nightly/struct.rs

```diff
@@ -23,15 +22,31 @@ fn main() {}
 // KnownLayout errors
 //
 
+struct NotKnownLayout;
+
+struct NotKnownLayoutDst([u8]);
+
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          N |        N |              N |        N |      KL00 |
 #[derive(KnownLayout)]
-struct KnownLayout1([u8]);
+struct KL00(u8, NotKnownLayoutDst);
 
-assert_impl_all!(KnownLayout1: KnownLayout);
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          N |        N |              Y |        N |      KL02 |
+#[derive(KnownLayout)]
+struct KL02(u8, [u8]);
 
+// | `repr(C)`? | generic? | `KnownLayout`? | `Sized`? | Type Name |
+// |          Y |        N |              N |        N |      KL08 |
 #[derive(KnownLayout)]
-struct KnownLayout2<T: ?Sized>(T);
+#[repr(C)]
+struct KL08(u8, NotKnownLayoutDst);
```

Comment:
Since this succeeds, should it be in a non-UI test file?

Reply:
This also has a failure:
```
error[E0277]: the trait bound `NotKnownLayoutDst: KnownLayout` is not satisfied
  --> tests/ui-msrv/struct.rs:41:10
   |
41 | #[derive(KnownLayout)]
   |          ^^^^^^^^^^^ the trait `KnownLayout` is not implemented for `NotKnownLayoutDst`
   |
   = help: see issue #48214
   = note: this error originates in the derive macro `KnownLayout` (in Nightly builds, run with -Z macro-backtrace for more info)
```